feat(n-input-number): Add hold event to change value of <n-input-numb…

NickyAlfEdge opened the pull request:
Functionality for hold click increment/decrement of the n-input-number icons, resolves: [#1293](https://github.com/TuSimple/naive-ui/issues/1293)

Have updated the test cases to accommodate for the changes, but I'm not overly sure if this is an ideal solution. It operates similarly to the approach implemented in Element Plus, albeit the existing implementation doesn't have an onleave event like Element Plus, so if the buttons are held down and then the mouse is moved away from them, then the number continues to increase/decrease (Though the n-input-number component also stays highlighted until it is unfocused anyway). In line with this, I clear the increase/decrease intervals onblur, this keeps in theme with how the inputs are in focus until other elements are focused.

07akioni commented on src/input-number/src/InputNumber.tsx:
```diff
@@ -457,12 +496,28 @@ export default defineComponent({
+                          <NBaseIcon
+                            clsPrefix={mergedClsPrefix}
+                            aria-disabled={true}
+                            onMousedown={
+                              !this.minusable ||
+                              this.mergedDisabled ||
+                              this.readonly
+                                ? undefined
+                                : this.handleMinusClick
+                            }
+                            onMouseup={
+                              !this.minusable ||
+                              this.mergedDisabled ||
+                              this.readonly
+                                ? undefined
+                                : this.clearMinusStepInterval
+                            }
```
This event should be bound to button, not icon

07akioni commented:
Increment is triggered even after mouse left button and up.

07akioni commented:
Increment should has a trigger threshold. For example only if you hold button after 1s, we start to increase the value.

NickyAlfEdge commented:
Have resolved the issues you mentioned.

- Added onMousedown, onMouseup and onMouseleave events to the NButton for use with this functionality.
- Moved the events to the parent button and not the icon.
- Added a 1 second delay prior to any incrementing/decrementing of the input value.
- Updated the test cases to reflect these changes.

Thanks so much for the comments, I was unsure whether to add anymore events to the existing Button. Anyway, I hope the latest changes will suffice, I tried to abide by the general approaches within the codebase.

NickyAlfEdge commented:
Never-mind, I've identified an issue. It doesn't seem to interact well with Min and Max props, will keep trying to figure this out.

NickyAlfEdge commented:
Apologies, the min / max prop issue should now be resolved. Any issues then please do highlight them and I will do my best to amend them.

07akioni commented:
I've refactor your commit. You can have a look at what should be changed in your PR.

758eed5d1ef6cfb0c6922c346c9e9908a86f09ad
